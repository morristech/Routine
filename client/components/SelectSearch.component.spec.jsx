/*
 * This file is part of the Routine project.
 *
 * (c) Yasser Ameur El Idrissi <getspookydev@gmail.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

import React from 'react';
import ReactDom from 'react-dom';
import { mount } from 'enzyme';
import SelectSearchComponent from '../components/SelectSearch.component';

let wrapper = null;
// fake data.
let faker = {
  defaultStateValue: 'Create-react-app',
  selectTemplate: jest.fn(() => {}),
  templates: [
    {
      name: 'Create-react-app',
      image: 'https://example.com/react.png',
      createdBy: 'Routine Team',
    },
  ],
};

beforeEach(function () {
  wrapper = (
    <SelectSearchComponent
      defaultStateValue={faker.defaultStateValue}
      selectTemplate={faker.selectTemplate}
      data={faker.templates}
    />
  );
});

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(wrapper, div);
  ReactDom.unmountComponentAtNode(div);
});

test('renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('check template default value', () => {
  const wrappedEnzyme = mount(wrapper);
  expect(wrappedEnzyme.prop('defaultStateValue')).toStrictEqual(
    faker.defaultStateValue,
  );
  expect(wrappedEnzyme.find('#element-name').at(0).text()).toEqual(
    faker.defaultStateValue,
  );
});
