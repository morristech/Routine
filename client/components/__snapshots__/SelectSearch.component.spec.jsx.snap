// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<SelectSearchComponent
  data={
    Array [
      Object {
        "createdBy": "Routine Team",
        "image": "https://example.com/react.png",
        "name": "Create-react-app",
      },
    ]
  }
  defaultStateValue="Create-react-app"
  selectTemplate={
    [MockFunction] {
      "calls": Array [
        Array [
          "Create-react-app",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
/>
`;
